### PR TITLE
Moved current tick incrementation outside of the if-initialized block

### PR DESCRIPTION
**Pull request by da-bao-jian**
for fn simulate_swap_mut_with_cache and fn simulate_swap_with_cache, the original implementation in [UniswapV3Pool.sol](https://github.com/Uniswap/v3-core/blob/d8b1c635c275d2a9450bd6a78f3fa2484fef73eb/contracts/UniswapV3Pool.sol#L725) will move the current tick no matter if the tick is initialized or not.

### CODE REVIEW

File: src/pool/mod.rs [hunk 1]
```diff
@@ -16,7 +16,7 @@ pub mod uniswap_v3;
 pub use uniswap_v2::UniswapV2Pool;
 pub use uniswap_v3::UniswapV3Pool;
 
-#[derive(Clone, Copy, Debug)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
```
Comment (0xOsiris) [inline src/pool/mod.rs:19]: Can you remove `PartialEq` & `Eq` traits. Imo these wouldn't have much utility in this context?

File: src/pool/uniswap_v2.rs [hunk 2]
```diff
@@ -18,7 +18,7 @@ pub const SYNC_EVENT_SIGNATURE: H256 = H256([
     199, 139, 229, 14, 6, 43, 3, 169, 255, 251, 186, 209,
 ]);
 
-#[derive(Debug, Clone, Copy, Default)]
+#[derive(Debug, Clone, Copy, Default, PartialEq, Eq)]
```
Comment (0xOsiris) [inline src/pool/uniswap_v2.rs:21]: Can you remove `PartialEq` & `Eq` here as well

File: src/pool/uniswap_v3.rs [hunk 3]
```diff
@@ -22,7 +22,7 @@ pub const SWAP_EVENT_SIGNATURE: H256 = H256([
 pub const U256_TWO: U256 = U256([2, 0, 0, 0]);
 pub const Q128: U256 = U256([0, 0, 1, 0]);
 pub const Q224: U256 = U256([0, 0, 0, 4294967296]);
-#[derive(Clone, Copy, Debug, Default)]
+#[derive(Clone, Copy, Debug, Default, PartialEq, Eq)]
```
Comment (0xOsiris) [inline src/pool/uniswap_v3.rs:25]: Here as well

**Review by 0xOsiris — CHANGES_REQUESTED** [review]
LGTM, just added a few comments. Can you also run `cargo fmt` and `cargo clippy --fix` so ci passes.

**Comment by 0xKitsune** [comment]
Thanks for making this PR! Once the requested changes are addressed, we can merge this in.

**Pull request merged**